packages/react/src/__tests__/events.test.tsx

```diff
@@ -51,13 +50,17 @@ it('should assign correct path to an event target', () => {
               key: 'container',
               props: {
                 ['data-testid']: 'button',
-                onClick: event(jest.fn()),
+                onClick: {
+                  $: 'event',
```

Comment:
I made types more strict. `event` returns `ServerEventDataState` (it has property "handler" that can be a function) and that property is obviously gone on the client after serialisation. So, we can't use that utility here, because it would make TypeScript complain.

This is client-side JSON definition passed directly to React component, so we have to make sure it does not have any server-specific properties (just like in real life).

Reply:
great, I know this stuff is tricky but it seems like a big step in the right direction. Definitely can't use the same state for client and server.